### DISCUSSION TITLE
PollEvented on a Windows Named Pipe in tokio 0.3

### DISCUSSION THREAD

**Discussion by fussybeaver** [tokio-rs/tokio#3112] [Q&A]
I see tokio 0.3.1 is now using an `AsyncFd` to replace the previous `PollEvented` API, but I'm using `PollEvented` to provide `AsyncRead` and `AsyncWrite` to a Windows Named Pipe in [bollard](https://github.com/fussybeaver/bollard). `AsyncFd` is unix-only - is there an plan for windows, or am I missing an API ?

**Comment by Darksonn** [comment]
We definitely want to support windows, but the details have not yet been figured out.

**Comment by carllerche** [comment]
I believe that the best strategy would be to just add a named pipe type in Tokio.
